src/System.Linq/src/System/Linq/AppendPrepend.cs

```diff
@@ -352,12 +378,36 @@ public override AppendPrependIterator<TSource> Prepend(TSource item)
                 return new AppendPrependN<TSource>(_source, _prepended != null ? _prepended.Add(item) : new SingleLinkedNode<TSource>(item), _appended);
             }
 
+            private TSource[] LazyToArray()
+            {
+                Debug.Assert(GetCount(onlyIfCheap: true) == -1);
+
+                var builder = new LargeArrayBuilder<TSource>(initialize: true);
+
+                for (SingleLinkedNode<TSource> node = _prepended; node != null; node = node.Linked)
+                {
+                    builder.Add(node.Item);
+                }
+
+                builder.AddRange(_source);
+
+                if (_appended != null)
+                {
+                    foreach (TSource item in _appended.ToArray())
```

Comment:
Was such an array getting allocated before?  I'm struggling to see this as a pure win if we're introducing additional intermediate ToArrays like this.  If it was already there before, then ok, though it would be nice subsequently to find a way to avoid it.